taipy/core/cycle/_cycle_manager.py

```diff
@@ -24,37 +24,55 @@
 
 class _CycleManager(_Manager[Cycle]):
 
-    _repository = _CycleRepository()
+    _repository: _CycleRepository = _CycleRepository()
     _ENTITY_NAME = Cycle.__name__
 
     @classmethod
     def _create(
-        cls, frequency: Frequency, name: str = None, creation_date: datetime = None, display_name=None, **properties
+        cls,
+        frequency: Frequency,
+        name: str = None,
+        creation_date: datetime = None,
+        display_name=None,
+        *args,
+        **properties
     ):
         creation_date = creation_date if creation_date else datetime.now()
-        start_date = _CycleManager._get_start_date_of_cycle(frequency, creation_date)
-        end_date = _CycleManager._get_end_date_of_cycle(frequency, start_date)
+        start_date = _CycleManager._get_start_date_of_cycle(frequency, creation_date, *args, **properties)
+        end_date = _CycleManager._get_end_date_of_cycle(frequency, start_date, *args, **properties)
         properties["display_name"] = display_name if display_name else start_date.isoformat()
+        cls._clean_properties(*args, **properties)
         cycle = Cycle(
             frequency, properties, creation_date=creation_date, start_date=start_date, end_date=end_date, name=name
         )
-        cls._set(cycle)
+        cls._set(cycle, *args, **properties)
         return cycle
 
+    @classmethod
+    def _clean_properties(cls, *args, **kwargs):
+        pass
+
```

Comment:
Putting this method without implementation is quite ambigious. How about we defined all the allowed properties for a cycle then filter it, instead of omitting the `token`? What do you think?

Reply:
We have implemented it as a dictionary so the user can have custom properties. Let's consider for example that the user wants to add a description, a creator name, or a list of user comments, ...